### PR TITLE
Feature/ft relative attention

### PR DESCRIPTION
**Pull request by wenshuoliu**
This PR add support to relative attention in saving/loading npz checkpoint; and fine-tuning in TF. It is tested with LookupTableEmbeddings + rpr_k, and the TF module output the same as pytorch one with random input. 

Note: need to re-run `ckpt2npz.py` to old ckpts after this is merged.

### CODE REVIEW

File: addons/embed_tlm_pytorch.py [hunk 1]
```diff
@@ -126,15 +126,13 @@ def __init__(self, **kwargs):
         self.d_model = int(kwargs.get('dsz', kwargs.get('d_model', 410)))
         d_ff = int(kwargs.get('d_ff', 2100))
         d_k = kwargs.get('d_k')
-        rpr_k = kwargs.get('rpr_k')
         embed_type = kwargs.get('word_embed_type', 'positional')
-        if embed_type == 'positional':
-            x_embedding = PositionalLookupTableEmbeddings(vsz=self.vsz, dsz=self.d_model)
-        elif embed_type == 'learned-positional':
-            x_embedding = LearnedPositionalLookupTableEmbeddings(vsz=self.vsz, dsz=self.d_model)
+        rpr_k = kwargs.get('rpr_k')
```
Comment (dpressel) [inline addons/embed_tlm_pytorch.py]: this was already merged in #478

File: addons/embed_tlm_pytorch.py [hunk 2]
```diff
@@ -126,15 +126,13 @@ def __init__(self, **kwargs):
...
+        x_embedding = create_embeddings(vsz=self.vsz, dsz=self.d_model, embed_type=embed_type)
         self.dsz = self.init_embed({'x': x_embedding})
         self.proj_to_dsz = pytorch_linear(self.dsz, self.d_model) if self.dsz != self.d_model else _identity
-        self.transformer = TransformerEncoderStack(num_heads, d_model=self.d_model, pdrop=pdrop, scale=True, layers=layers, d_ff=d_ff, rpr_k=rpr_k, d_k=d_k)
+        self.transformer = TransformerEncoderStack(num_heads, d_model=self.d_model, pdrop=pdrop, scale=True,
```
Comment (dpressel) [inline addons/embed_tlm_pytorch.py]: Already merged in #478

File: addons/embed_tlm_pytorch.py [hunk 3]
```diff
@@ -126,15 +126,13 @@ def __init__(self, **kwargs):
...
         d_ff = int(kwargs.get('d_ff', 2100))
         d_k = kwargs.get('d_k')
-        rpr_k = kwargs.get('rpr_k')
         embed_type = kwargs.get('word_embed_type', 'positional')
-        if embed_type == 'positional':
-            x_embedding = PositionalLookupTableEmbeddings(vsz=self.vsz, dsz=self.d_model)
-        elif embed_type == 'learned-positional':
-            x_embedding = LearnedPositionalLookupTableEmbeddings(vsz=self.vsz, dsz=self.d_model)
+        rpr_k = kwargs.get('rpr_k')
+        x_embedding = create_embeddings(vsz=self.vsz, dsz=self.d_model, embed_type=embed_type)
```
Comment (dpressel) [inline addons/embed_tlm_pytorch.py]: why is this change required?

Comment (wenshuoliu) [reply addons/embed_tlm_pytorch.py]: when we use `rpr_k`, there's no need to use positional embeddings any more. This change enables using `default` + `rpr_k`.
